calyx-py/calyx/gen_exp.py

```diff
@@ -673,36 +964,145 @@ def generate_exp_taylor_series_approximation(
     parser.add_argument("-w", "--width", type=int)
     parser.add_argument("-i", "--int_width", type=int)
     parser.add_argument("-s", "--is_signed", type=bool)
+    parser.add_argument("-e", "--base_is_e", type=bool)
 
     args = parser.parse_args()
 
-    degree, width, int_width, is_signed = None, None, None, None
-    required_fields = [args.degree, args.width, args.int_width, args.is_signed]
+    degree, width, int_width, is_signed, base_is_e = None, None, None, None, None
+    required_fields = [args.degree, args.width,
+                       args.int_width, args.is_signed, args.base_is_e]
     if all(map(lambda x: x is not None, required_fields)):
         degree = args.degree
         width = args.width
         int_width = args.int_width
         is_signed = args.is_signed
+        base_is_e = args.base_is_e
     elif args.file is not None:
         with open(args.file, "r") as f:
             spec = json.load(f)
             degree = spec["degree"]
             width = spec["width"]
             int_width = spec["int_width"]
             is_signed = spec["is_signed"]
+            base_is_e = spec["base_is_e"]
     else:
         parser.error(
             "Need to pass either `-f FILE` or all of `-d DEGREE -w WIDTH -i INT_WIDTH`"
         )
 
+    # build 2 separate programs: 1 if base_is_e is true, the other if false
+    # any_base_program is (obviously) the one for any base
```

Comment:
Doesn't `any_base_program` also include `e`? Why did you make this decision?

Reply:
The way we calculate b^x (for any base b) is by doing e^ln(b^x). So, we need to generate a lot of extra components to do the ln(b^x). But if the base is just `e` then we don't need all those extra components.

Reply:
Got it. I would suggest adding a comment to clarify this. Alternatively just use 'base_is_e' and 'not base_is_e'. Since any base seems like a misnomer. Up to you :)

Reply:
On second thought, can we infer the base is 'e' when no base is provided, instead of adding an additional flag?

Reply:
I think it'll be easier to clear this up on a subsequent PR (I'll make it later today).